com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/cubrid/trans/converter/NumericConverter.java

```diff
@@ -46,6 +48,41 @@
 public class NumericConverter extends
 		AbstractDataConverter {
 
+	/**
+	 * If it is NUMERIC type and there is a decimal point,
+	 * it returns the rounded value if the total number of digits exceeds 38.
+	 * 
+	 * @param obj Object
+	 * @param dataType String
+	 * @return value Object
+	 */
+
+	private Object getCUBRIDDataSet(Object obj) {
+		Object value = obj;
+		int maxSize = DataTypeConstant.NUMERIC_MAX_PRECISIE_SIZE ;
+		String strValue = obj.toString() ;
+		if (strValue.charAt(0) == '.') {
+			if (strValue.length() > maxSize + 1) {
+				value = new BigDecimal(obj.toString()).setScale(maxSize, RoundingMode.HALF_UP);
+			}
+		}
+		else if (strValue.charAt(0) == '0' && strValue.charAt(1) == '.') {
+			if (strValue.length() > maxSize + 2) {
+				value = new BigDecimal(obj.toString()).setScale(maxSize, RoundingMode.HALF_UP);
+			}
+		}
+		else {
+			if (strValue.length() > maxSize+1) {
+				int index = strValue.indexOf(".");
+				if (index > 0 && maxSize >= index) {
+					value = new BigDecimal(obj.toString()).setScale(maxSize-index, RoundingMode.HALF_UP);
```

Comment:
if we have index 1, it means we have a number like 1.0123, and 'maxSize' is 38 then calculate scale is 37. Then setscale (37, ...) will generate a number of 37 decimal place. Then we will get 1 + 1 + 37 digit number, it exceed maximum precision. I think we 'maxSize-index-1' as a SCALE of new number because of decimal point itself.

My suggestion from #64 to #81 is:

int index = strValue.IndexOf(".");
if ((index >= 0 && index < maxSize) && strValue.length() > maxSize) {
	value = new BigDecimal(obj.toString()).setScale(maxSize - index -1, RoundingMode.HALF_UP);
}

Reply:
Thank you for your comment.

I think the '-1' seems to be not needed in the source you told me.

Because, if the number start with a '.', we think the scale should be determined as '38 - 0 = 38 '. Also, if there is more than one integer in front, we think that scale should be decided like '38 - 1 = 37', '38 - 2 = 36', ...

Also, the test results seem to match the reply I posted.

Finally, I feel that you are asking me to write the source as briefly as possible while watching the source, so I will modify the source and test it.

Thank you.